packages/uhk-web/src/app/components/side-menu/side-menu.component.html

```diff
@@ -144,8 +144,12 @@
                     Macros
                     <button (click)="addMacro()"
                             class="btn btn-sm btn-secondary float-right"
-                            [class.disabled]="state.updatingFirmware">
-                        <fa-icon [icon]="faPlus"></fa-icon>
+                            [disabled]="state.updatingFirmware || state.macros.length > maxAllowedMacros">
+                        <fa-icon [icon]="faPlus"
+                                 [ngbTooltip]="state.macros.length > maxAllowedMacros ? 'No more than 255 macros are supported.' : null"
```

Comment:
Make the string contain ${maxAllowedMacros} to avoid duplication and set its value to 254.

Reply:
done. And I replaced the `>` condition to `>=`